map.h

```diff
@@ -0,0 +1,252 @@
+                    break;
+                case Map::Wall:
+                    std::cout << "██";
+                    break;
+                case Map::Her:
+                    std::cout << "┣┫";
+                    break;
+                case Map::Monst:
+                    if(level.multMonsterPresent(i,j))
+                        std::cout << "MM";
+                    else
+                        std::cout << "M░";
+                    break;
+                default:
+                    break;
+                }
+            }
+            std::cout << "║";
+            std::cout << "\n";
+        }
+        std::cout << "╚═";
+        for(int i=1; i<level.getWidth(); i++) //first row
+            std::cout << "══";
+        std::cout << "═╝\n";
+    }
+
+    void moveHero(int xc, int yc){
+        int xx = hero_pos.x+xc;
+        int yy = hero_pos.y+yc;
+        if(level.get(xx,yy) != Map::Wall && 0<=xx && xx<=level.getWidth() && 0<=yy && yy<=level.getHeight()){
+            level.put(hero_pos.x,hero_pos.y,' ');
+            hero_pos.x = xx;
+            hero_pos.y = yy;
+            level.put(hero_pos.x,hero_pos.y,'H');
+        }
+        std::list<Monster*> m;
+        for (auto i = monster_list.cbegin(); i != monster_list.cend(); ) {
+            if (i->second.x == hero_pos.x && i->second.y == hero_pos.y){
+                m.push_back(i->first);
+                her->fightTilDeath(*i->first);
+                if (!i->first->isAlive()){
+                    std::cout << i->first->getName() <<" monster died\n";
+                    monster_list.erase(i++);
+                }else{
+                    ++i;
+                }
+            }else{
+                    ++i;
+                }   
+        }
+    }
+public:
+    Game(){};
+    Game(std::string mapfilename) { // Game with the Map initialized
+        level = Map(mapfilename);
+        level_given = true;
+    }
+    void setMap(Map map){ // Set the map
+        if (her!=nullptr && monster_list.empty()) throw Game::AlreadyHasUnitsException();
+        level = map;
+        level_given = true;
+    }
+    void putHero(Hero hero, int x, int y){
+        if (!level_given) throw Map::WrongIndexException();
+        if (her!=nullptr) throw Game::AlreadyHasHeroException();
+        if (runing) throw Game::GameAlreadyStartedException();
+        her = new Hero(hero);
+        hero_pos = posit(x,y);
+        level.put(x,y,'H');
+        can_be_run = true;
+    }
+    void putMonster(Monster monster, int x, int y) {
+        if (!level_given) throw Map::WrongIndexException();
+        if (runing) throw Game::GameAlreadyStartedException();
+        level.put(x,y,'M');
+        monster_list.insert(std::pair<Monster*,posit>(new Monster(monster),posit(x,y)));
+    }
+
+    void run(){
+        if(runing==true) ;
+        if(!level_given || her == nullptr) throw Game::NotInitializedException();
+        std::map<Monster*, posit> m_list = monster_list;
```

Comment:
ez csak egy shallow copy-t csinal, ugye tudod? Azaz a pointer kulcsokat lemasolja, de a monstereket nem.

Reply:
De nem is teljesen ertem, miet csinalsz masolatot.

Reply:
Itt a logika az lett volna, hogy ha mondjuk egyszer lefut a run() akkor a monsterek akik közben meghaltak ne veszenek el hanem kerüljenek újra vissza, bár valóban ezt egy shallow copy nem valósítja meg.